pkg/controller/sshd/sshd_controller.go

```diff
@@ -358,45 +367,16 @@ func (r *ReconcileSSHD) Reconcile(request reconcile.Request) (reconcile.Result,
 		}
 	}
 
-	// Create a Route53 DNS entry for the service's load balancer.
-	// TODO Consider using https://github.com/openshift/external-dns
-
-	clusterBaseDomain, err := utils.GetClusterBaseDomain(r.client)
-	if err != nil {
-		r.SetSSHDStatusError(instance, "Failed to get cluster's base domain", err)
-		return reconcile.Result{}, err
-	}
-
-	// Get the ELB-to-be's name from Service's UID.
-	elbName := strings.ReplaceAll("a"+string(foundService.ObjectMeta.UID), "-", "")
-	if len(elbName) > 32 {
-		// truncate to 32 characters
-		elbName = elbName[0:32]
-	}
-
-	exists, elb, err := r.awsClient.DoesELBExist(elbName)
-	if err != nil {
-		r.SetSSHDStatusError(instance, "Failed to get load balancer status", err)
-		return reconcile.Result{}, err
-	}
-	if !exists {
-		// It isn't bad that it doesn't exist if there's no error, so re-queue.
-		r.SetSSHDStatusPending(instance, "Waiting on service load balancers")
+	err = r.cloudClient.EnsureSSHDNS(context.TODO(), r.client, instance, service)
+	switch err {
+	case nil:
+		// all good
+	case err.(*cioerrors.LoadBalancerNotFoundError):
+		// couldn't find the new ELB yet
+		r.SetSSHDStatus(instance, "Couldn't reconcile", "AWS ELB isn't ready yet.")
```

Comment:
nit: This should be worded less cloud-specifically.  Maybe: `"Load balancer isn't ready yet"`